lib/Dialect/FIRRTL/FIRRTLOps.cpp

```diff
@@ -4781,6 +4785,16 @@ LogicalResult RWProbeOp::verifyInnerRefs(hw::InnerRefNamespace &ns) {
   return checkFinalType(symOp.getTargetResult().getType(), symOp.getLoc());
 }
 
+FIRRTLType RWProbeSSAOp::inferReturnType(ValueRange operands,
+                                         ArrayRef<NamedAttribute> attrs,
+                                         std::optional<Location> loc) {
+  auto type = operands[0].getType();
+  auto forceableType = firrtl::detail::getForceableResultType(true, type);
+  if (!forceableType)
+    return emitInferRetTypeError(loc, "cannot force type ", type);
```

Comment:
Is this check unnecessary given that the type is ODS-declared with the `RWProbeTarget` constraint?

Reply:
Not unnecessary but that could/should be fixed-- regardless since the method is able to fail seems good to check it.
Not 100% this is only called in places where ODS already checked things -- such as when updating types in type inference code and sometimes in the parser.